src/celldega/sp/__init__.py

```diff
@@ -0,0 +1,258 @@
+"""
+Module for spatial operation and analysis
+"""
+
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+import matplotlib.cm as cm
+import geopandas as gpd
+import pandas as pd
+
+def calc_distance_to_polygon(
+    gdf_polygons: gpd.GeoDataFrame,
+    gdf_points: gpd.GeoDataFrame,
+    roi_name: str
+) -> tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """
+    Calculate the distance of points to the specified polygon.
+    
+    Args:
+        gdf_polygons: GeoDataFrame containing polygons with a 'roi' column.
+        gdf_points: GeoDataFrame containing points to calculate distances for.
+        roi_name: Name of the region of interest (ROI) to calculate distances to.
+    
+    Returns:
+        A tuple containing:
+            - gdf_points: The input gdf_points with added distance column.
+            - gdf_polygon: The extracted polygon for the specified ROI.
+    """
+
+    print (f"Extract the polygon for the specified ROI: {roi_name}")
+    # Extract the polygon for the specified ROI
+    gdf_polygon = gdf_polygons.loc[gdf_polygons["roi"] == roi_name].reset_index()
+    
+    # Calculate distances from points to the polygon
+    gdf_points[f"distance_to_polygon_{roi_name}"] = gdf_points["geometry"].apply(
+        lambda point: gdf_polygon.geometry.iloc[0].distance(point)
+    )
+    print (f"Distances from points to polygon {roi_name} were calculated.")
+    return gdf_points, gdf_polygon
```

Comment:
I think we can no return `gdf_polygon` since this GeoDataFrame is not being updated.